central/role/resources/list.go

```diff
@@ -24,56 +24,92 @@ import (
+	// Integration is the new  resource grouping all integration resources.
+	Integration                      = newResourceMetadata("Integration", permissions.GlobalScope)
 	Indicator                        = newResourceMetadata("Indicator", permissions.NamespaceScope)
 	K8sRole                          = newResourceMetadata("K8sRole", permissions.NamespaceScope)
 	K8sRoleBinding                   = newResourceMetadata("K8sRoleBinding", permissions.NamespaceScope)
 	K8sSubject                       = newResourceMetadata("K8sSubject", permissions.NamespaceScope)
-	Licenses                         = newResourceMetadata("Licenses", permissions.GlobalScope)
 	Namespace                        = newResourceMetadata("Namespace", permissions.NamespaceScope)
-	NetworkBaseline                  = newResourceMetadata("NetworkBaseline", permissions.NamespaceScope)
 	NetworkGraph                     = newResourceMetadata("NetworkGraph", permissions.NamespaceScope)
-	NetworkGraphConfig               = newResourceMetadata("NetworkGraphConfig", permissions.GlobalScope)
 	NetworkPolicy                    = newResourceMetadata("NetworkPolicy", permissions.NamespaceScope)
 	Node                             = newResourceMetadata("Node", permissions.ClusterScope)
-	Notifier                         = newResourceMetadata("Notifier", permissions.GlobalScope)
 	Policy                           = newResourceMetadata("Policy", permissions.GlobalScope)
-	ProbeUpload                      = newResourceMetadata("ProbeUpload", permissions.GlobalScope)
-	ProcessWhitelist                 = newResourceMetadata("ProcessWhitelist", permissions.NamespaceScope)
-	Risk                             = newResourceMetadata("Risk", permissions.NamespaceScope)
-	Role                             = newResourceMetadata("Role", permissions.GlobalScope)
-	ScannerBundle                    = newResourceMetadata("ScannerBundle", permissions.GlobalScope)
-	ScannerDefinitions               = newResourceMetadata("ScannerDefinitions", permissions.GlobalScope)
 	Secret                           = newResourceMetadata("Secret", permissions.NamespaceScope)
-	SensorUpgradeConfig              = newResourceMetadata("SensorUpgradeConfig", permissions.GlobalScope)
 	ServiceAccount                   = newResourceMetadata("ServiceAccount", permissions.NamespaceScope)
-	ServiceIdentity                  = newResourceMetadata("ServiceIdentity", permissions.GlobalScope)
-	SignatureIntegration             = newResourceMetadataWithFeatureFlag("SignatureIntegration", permissions.GlobalScope, features.ImageSignatureVerification)
-	User                             = newResourceMetadata("User", permissions.GlobalScope)
 	VulnerabilityManagementApprovals = newResourceMetadata("VulnerabilityManagementApprovals", permissions.GlobalScope)
 	VulnerabilityManagementRequests  = newResourceMetadata("VulnerabilityManagementRequests", permissions.GlobalScope)
```

Comment:
It will be great if we could simplify it as it require non trivial access check and we need to do a workaround for this case in PG sac


```
 git grep "PermissionChecker interface"
central/clusterinit/store/postgres/permission_checker.go:type PermissionChecker interface {
central/integrationhealth/store/postgres/permission_checker.go:type PermissionChecker interface {
central/vulnerabilityrequest/datastore/internal/store/postgres/permission_checker.go:type PermissionChecker interface {
```

Reply:
Maybe we can add something to the generator to have an alternative resource as command line parameter, and create the ScopeChecker accordingly (as being `sac.ForResources(sac.ForResource(mainResource),sac.ForResource(altResource))` and use `AllowedToAny` in that case.